src/main/java/ee/tuleva/onboarding/auth/ocsp/OCSPRequest.java

```diff
@@ -0,0 +1,155 @@
+        return response;
+    }
+
+
+    private String validateOCSPResponse(OCSPResp response) throws OCSPException {
+
+        String status = "UNABLE_TO_TEST_USER_CERTIFICATE";
+        switch (response.getStatus()) {
+            case 0:
+                BasicOCSPResp ocspResponseData = (BasicOCSPResp) response.getResponseObject();
+                SingleResp[] responses = ocspResponseData.getResponses();
+                for (SingleResp response1 : responses) {
+                    if (response1.getCertStatus() == null) {
+                        status = "GOOD";
+                    } else if (response1.getCertStatus() instanceof RevokedStatus) {
+                        status = "CERTIFICATE_REVOKED";
+                    }
+                    log.info("OCSP response code: {0}", status);
+                }
+                break;
+            case 1:
+                log.warn("Malformed request. OCSP response code: {0}", response.getStatus());
+                throw new AuthenticationException(AuthenticationException.Code.UNABLE_TO_TEST_USER_CERTIFICATE, "Malformed request", new CertificateException());
+            default:
+                log.warn( "Uncaught error. OCSP response code: {0}", response.getStatus());
+                throw new AuthenticationException(AuthenticationException.Code.UNABLE_TO_TEST_USER_CERTIFICATE, "Uncaught error", new CertificateException());
+        }
+        return status;
+    }
+
+
+    private OCSPReq generateOCSPRequest(BigInteger serial) {
+
+        OCSPReq request = null;
+        try {
+            CertificateFactory certFactory = CertificateFactory.getInstance("X.509");
+            InputStream in = new ByteArrayInputStream(ca.getBytes("UTF-8"));
+            X509Certificate issuerCert = (X509Certificate) certFactory.generateCertificate(in);
+
+            JcaDigestCalculatorProviderBuilder digestCalculatorProviderBuilder = new JcaDigestCalculatorProviderBuilder();
+            DigestCalculatorProvider digestCalculatorProvider = digestCalculatorProviderBuilder.build();
+            DigestCalculator digestCalculator = digestCalculatorProvider.get(CertificateID.HASH_SHA1);
+
+            log.info( "CA subject: {0}", issuerCert.getIssuerX500Principal().getName("CANONICAL"));
+            log.info("Serial number: {0}", serial);
+
+            CertificateID id = new CertificateID(digestCalculator, new JcaX509CertificateHolder(issuerCert), serial);
+            OCSPReqBuilder ocspGen = new OCSPReqBuilder();
+            ocspGen.addRequest(id);
+
+            request = ocspGen.build();
+        } catch (CertificateException | OCSPException | OperatorCreationException | UnsupportedEncodingException e) {
+            throw new AuthenticationException(AuthenticationException.Code.UNABLE_TO_TEST_USER_CERTIFICATE, "Uncaught error", e);
+        }
+
+        return request;
+    }
+
+    private OCSPResp sendPost(byte[] request) throws IOException {
```

Comment:
Can you do this using a RestTemplate instead? Looks like it is really hard to test